include/GcInfo/GcInfo.h

```diff
@@ -29,46 +29,103 @@
 class GcInfoAllocator;
 class GcInfoEncoder;
 
-/// \brief Per Function GcInfo
+// Allocation type information for a stack allocation.
+//
+// While some of the following information is deducible
+// from the Alloca instruction (ex: GcPointer/Aggregate)
+// we cache the information in the flag bits instead of
+// walking through the type every time.
+
+enum AllocaFlags {
+  Alloca_None = 0x0,
+  Alloca_GcPointer = 0x1,
+  Alloca_GcAggregate = 0x2,
+  Alloca_Pinned = 0x4,
+  Alloca_GsCookie = 0x8,
+  Alloca_SecurityObject = 0x10,
+  Alloca_GenericsContext = 0x20
+};
```

Comment:
Since there's already `Alloca` in the enum name, do we need to repeat it in each enumerate?

I think `AllocaFlags::Pinned` reads a bit better.